### PR TITLE
[DF] ROOT-9470 - Optimise GetBranchNames saving them in the RInterface

### PR DESCRIPTION
**Pull request by dpiparo**

### CODE REVIEW

File: tree/dataframe/inc/ROOT/RDFInterface.hxx [hunk 1]
```diff
@@ -1726,9 +1724,6 @@ private:
 
       RInterface<RLoopManager> cachedRDF(std::make_shared<RLoopManager>(std::move(ds), columnList));
 
-      const std::vector<std::string> columnTypeNames = {RDFInternal::TypeID2TypeName(
-         typeid(typename std::decay<decltype(std::get<S>(colHolders))>::type::Value_t))...}; // ... expands on S
-
```
Comment (bluehood) [inline tree/dataframe/inc/ROOT/RDFInterface.hxx]: this change is unrelated, right?

File: tree/dataframe/inc/ROOT/RDFInterface.hxx [hunk 2]
```diff
@@ -111,6 +111,7 @@ class RInterface {
    ColumnNames_t fValidCustomColumns; ///< Names of columns `Define`d for this branch of the functional graph.
    /// Non-owning pointer to a data-source object. Null if no data-source. RLoopManager has ownership of the object.
    RDataSource *const fDataSource = nullptr;
+   std::shared_ptr<const ColumnNames_t> fDatasetColumns; ///< Cache of the chain columns names
```
Comment (bluehood) [inline tree/dataframe/inc/ROOT/RDFInterface.hxx]: If I understand correctly the variable should contain just the TTree columns, if any, not all the columns. And it's all the possible variations of valid column names, so each column is in there multiple times...maybe `fValidBranchNames`?

**Comment by bluehood** [comment]
@dpiparo for per validation, see https://sft.its.cern.ch/jira/browse/ROOT-9470

File: tree/dataframe/inc/ROOT/RDFInterface.hxx (hunk 2 above)
Comment (dpiparo) [reply tree/dataframe/inc/ROOT/RDFInterface.hxx]: That's a good point. I was not sure what name to use and if the name should avoid ttree jeargon. Since this is really about trees and is a private data member, we can probably use branch rather than column.
